openfisca_uk/variables/finance/benefit/family/tax_credits.py

```diff
@@ -266,9 +280,9 @@ def formula(benunit, period, parameters):
         takeup_rate = parameters(
             period
         ).benefit.tax_credits.working_tax_credit.takeup
-        return benunit("claims_legacy_benefits", period) & (
-            random(benunit) < takeup_rate
-        ) | benunit("claims_all_entitled_benefits", period)
+        takes_up = random(benunit) < takeup_rate
+        would_take_up = benunit("claims_legacy_benefits", period) & takes_up
+        return would_take_up | benunit("claims_all_entitled_benefits", period)
```

Comment:
@nikhilwoodruff could you confirm that this is the intended logic? `x & y | z` is sort of ambiguous as written (without parens).

Reply:
Yes, this preserves the intention: claims all should override everything.